[18.0][MIG] web_responsive: Migration to 18.0

kobros-tech opened the pull request.

lasley committed:
[FIX] web_responsive: Other minor fixes
* Add Tecnativa as author
* Remove untestable JS method
* Rename to web_responsive

lasley committed:
[MIG] web_responsive: Migrate to v10
* Bump version
* Change all openerp to odoo
* Fix qunit suite injection
* Remove jQuery require
* Change to new selectors:
  * `oe_leftbar` to `o_sub_menu`
  * `oe_logo` to `o_sub_menu_logo`
  * `oe_footer` to `o_sub_menu_footer`
  * `oe_secondary_menus_container` to `o_sub_menu_content`
* Add style to hide oe_footer instead of remove to not break `support_branding`
* Add note in ReadMe explaining override of `support_branding`
* Set top margin of app drawer title to 0 to fit v10 proportions
* Fix notification badge positioning
* Add o_web_client class to body to fix overlaps
* Scroll control panel with page
* Change navbar z-index to not be overlapped by buttons
* Raise z-index on header to raise over buttons
* Handle layered notifications via z-index
* Remove `#` from navigation links HREF to prevent history littering

lasley committed:
[FIX] web_responsive: Change z-indexes
* Rebalance z-indexes to fix overlay issue while still providing notification support

oca-transbot committed:
Update translation files

danieltorres7 committed:
Form view improvements to fit smaller screens

On screens with less than 768px content is limited to 2 columns.

On mobile field labels have their own line.

Fix elements width for XS screens

danieltorres7 committed:
Form image editing controls always present

Make image editing controls always available, instead of depending
on resolution or hover.

danieltorres7 committed:
Web responsive menu improvements (#513)

* Bold titles for apps in the app-drawer
* Remove z-index from Control Panel buttons, so it doesn't overlap the menu.
* Better responsive menus
  * Move systray icons/menus out of the menu and up to the top bar.
  * Increase avatar size and visibility.

* Other minor menu behavior fixes.
* Changes to fit odoo variables
* And a little margin adjust on navbar-right
* Top menu always thick
* Removed overflow:hidden!important from main.less
* Local patch: prevent body overflow auto
* Stop using `style=` on `<body>`
* Fix jquery drawer in anticipation of:
   https://github.com/blivesta/drawer/pull/36

yajo committed:
Fix problem with overflowing chatter widget.

yajo committed:
[FIX][web_responsive] Form padding, titles and buttons OK in XS

Small makeup to make it look good in XS screens.

sergio-teruel committed:
[10.0][IMP] web_responsive: Load default kanban view for small screen

Commit:
[10.0][IMP][web_responsive] (#569)

* [IMP] drawer-toggle

* [IMP] Hitbox of close drawer-toggle

* [IMP] Removed outline on menu item

* [IMP] Logo positionings

* [FIX] Try me on runbot button

* [REM] Reversed outline change

zeroheure committed:
Update README to fix wikipedia link

lasley committed:
[FIX] web_responsive: Change edit accesskey
* Change accesskey for `edit` in form view back to `e` to fix #587

yajo committed:
[FIX][web_responsive] Make it compatible with l10n_es_toponyms

Before this patch, there was an incompatibility between this addon and l10n_es_toponyms, caused by the lack of wrapping of the special address field that was being added.

Now layout works as expected in that case. In any other case, it seems sensible to wrap inputs so no more weird overflows happen and less chances of needing horizontal scrolling happen.

pedrobaeza committed:
[FIX] web_responsive: Execution permission

Commit:
[10.0][IMP][web_responsive] Make search bar sticky  (#633)

* Initial commit

* Initial commit

yajo committed:
[FIX][web_responsive] Long-named multicompanies nice in XS (#688)

lasley committed:
[FIX] web_responsive: Chrome Slow Scrolling
* Update to iScoll bleeding edge to fix #741

angelmoya committed:
UPDT version

angelmoya committed:
FIX travis (#747)

* FIX hasclass

* FIX views

* [FIX] web_responsive: Syntax error in xpath

pedrobaeza committed:
[FIX] web_responsive: Correct runbot link

hgminh95 committed:
[MIG] web_responsive test to 11.0

lasley committed:
[FIX] web_responsive: Menu Height on Init
* Add a call to the window resize method during init to fix OCA#773

fanha99 committed:
[FIX] web_responsive: Hide header when print directly from browser issues 808 (#809)

Closes #808

lasley committed:
[IMP] web_responsive: Add search feature (#838)

* Add feature to search menus in the app drawer
* Switch model to RPC

Tardo committed:
[IMP] web_responsive: Possibility of putting the chatter in the right  (#951)

OCA-git-bot committed:
[UPD] README.rst

Tardo committed:
[FIX] Chatter Position

Tardo committed:
[FIX] Read/Write permissions

Tardo committed:
[CLEAN] Comments

Tardo committed:
[IMP] Code

Tardo committed:
[IMP] Adding res.users chatter_position read/write test

yajo committed:
Fix linter

Tardo committed:
[IMP] Sticky Statusbar

Tardo committed:
[IMP] Add sticky header & footer in list view

Tardo committed:
[FIX][11.0][web_responsive] Views without sheet

yajo committed:
[FIX] web_responsive: Complete search feature forward-porting

Tardo committed:
[IMP][11.0][web_responsive] Suggested changes (#978)

* [IMP] Chatter Position
* [IMP] Clean unused stuff
* [IMP] FieldStatus hidden-xs
* [IMP] Improvements suggested in #540
* [WIP] Suggested changes
* [WIP] Improve FieldStatus-Responsive
* [IMP] Statusbar Buttons & FieldStatus
* [IMP] Statusbar Buttons Style
* [IMP] Button 'Task' style

yajo committed:
[FIX] web_responsive: Disable access to hidden menus

When doing menu search, some menus that the user didn't have permission to use were presented to him.

Fix #850.

pedrobaeza committed:
[IMP] web_responsive: Add known issue note about sticky

OCA-git-bot committed:
[UPD] README.rst

OCA-git-bot committed:
[UPD] README.rst

yajo committed:
[MIG] web_responsive: Migrate to v12 and refactor

This migration includes a full refactoring to make this module more
maintainable. Some things that have changed:

- Removed external libraries.
- Change Less for Scss.
- Reduce ES and XML to the minimal required needs.
- Implement as much features as possible with just Scss.
- Remove copyright from `__init__.py` files.
- Trigger the new hotkeys system from Odoo v12 with `Shift+Alt` instead
  of just `Alt`, and restore some good old hotkeys (`E` for "Edit",
  `D` for "Discard", and `A` for "Apps menu").
  See https://github.com/odoo/odoo/issues/30068 on the matter.
- Control panel breadcrumbs are collapsed into a single backwards icon.
- Add FA icons to most common buttons in control panel.
- Hide text in XS for those buttons, to have a slicker phone experience.
- Lots of gifs in the README!

oca-travis committed:
Update translation files

yajo committed:
[FIX] web_responsive: Skip re-search when not writing

When the user uses TAB or SHIFT+TAB to navigate in search results, he can trigger a SHIFT-only keydown event without noticing, which would reset the selected result position to the first or last.

It is disturbing, so, to avoid that problem, the re-search is only triggered if the currently pressed key has length=1 or is Backspace, which will skip most keys that are not actually writing a character into the search input, like i.e. "Shift", "Alt", "F4", etc.

In addition to that, to detect if the search results is empty, the `:empty` selector is not trustable because it considers not empty nodes with whitespace-only content. That has been patched too.

yajo committed:
[FIX] web_responsive: Make app icons and search results middle-button-clickable

In the case of app icons, those weren't getting the `href` attribute filled.

In the case of search results, those weren't even an `<a>` element.

Fixed in both cases to fix https://github.com/OCA/web/issues/1170.

Tardo committed:
[IMP] Sticky chatter topbar

Tardo committed:
[FIX+IMP] web_responsive: Chatter Topbar + fix #1189 + added gif

yajo committed:
Update web_responsive/static/src/css/web_responsive.scss

Co-Authored-By: Tardo <dev@redneboa.es>

yajo committed:
[FIX] web_responsive: Allow searching apps in mobile Chrome

This patch fixes #1251.

The keydown event is faked in mobile chrome because it's not something reliable due to today's virtual keyboards prediction features. Typing a <kbd>C</kbd> character doesn't mean it will be there when user uses autocorrection/autoguessing.

Instead of that, we use now the `keydown` event exclusively for movement features, and `input` event exclusively for content changes in the search input element.

Apart from fixing search in mobile Chrome, it also makes code more readable.

The search input also disables autocompletion since it would be useless and annoying here.

Tardo committed:
[IMP] web_responsive: chatter width

Tardo committed:
[IMP] web_responsive: Sticky in tree view and form view

Tardo committed:
[IMP] web_responsive: Support long title in form view

Tardo committed:
[IMP] web_responsive: AppMenu wait for load view

Tardo committed:
[IMP] web_responsive: Suggested changes

Tardo committed:
[IMP] web_responsive: Menu wait for action

Tardo committed:
[IMP] web_responsive: Readme

Tardo committed:
[FIX] web_responsive: AppDrawer in Chromium

Tardo committed:
[IMP] web_responsive: Refactor code

Tardo committed:
[IMP] web_responsive: Readme

Tardo committed:
[FIX] web_responsive: Chatter position

Tardo committed:
imp overflow

Tardo committed:
[IMP] web_responsive: Chatter on small screen

Tardo committed:
[FIX] web_responsive: Close AppDrawer

keshrath committed:
[FIX] web_responsive: AppDrawer Discard

keshrath committed:
[FIX] web_responsive: adapt AppDrawer Discard

keshrath committed:
[FIX] web_responsive: Close Menu on Discard

OCA-git-bot committed:
[UPD] README.rst

keshrath committed:
Update web_responsive.scss

Tardo committed:
[IMP] web_responsive: Performance when hide menus

Tardo committed:
[IMP] web_responsive: Don't show menu if waiting for an action to end

Tardo committed:
[IMP] web_responsive: Small refactor 'isWaiting'

Tardo committed:
[FIX] web_responsive: changed button box more initial position

Tardo committed:
[MIG] web_responsive: Migration to 13.0

Tardo committed:
[IMP] web_responsive: Made chatter input & attachments scrollable

Tardo committed:
[IMP] web_responsive: Support long titles

Tardo committed:
black, isort

pedrobaeza committed:
[IMP] web_responsive: Add development status + contributors

OCA-git-bot committed:
[UPD] README.rst

emagdalenaC2i committed:
[FIX] web_dialog_size js warnings

Tardo committed:
[IMP] web_responsive: Adapt test to 13.0

Tardo committed:
[IMP] mail_tracking: Changed form statusbar z-index

Tardo committed:
[IMP] web-responsive: Document Viewer

Commit:
[FIX] web_responsive: make followers dropdown usable

Fix a visual glitch that impeded using the followers dropdown with a sided chatter.

Commit:
[IMP] pre-commit run after update

Includes some manual fixes to silent ESLint warnings.

Tardo committed:
[FIX] web_responsive: Change AppDrawer close method

Previous of this commit, AppDrawer closes when the action have an id
that is defined in a menu item and no rpc calls are running.
Now AppDrawer closes in a more generic way, is enough that a
controller is attached to consider that the AppDrawer needs to be
closed.

sbidoul committed:
pre-commit update: prettier xml indent

kos94ok-3D committed:
[FIX] web_responsive: fix hotkey combinations in a shortcut template

Tardo committed:
[IMP] web_responsive: Restore discuss mobile mode

androettop committed:
New style in apps menu

scss file to change ui variables easily

Deleted unnecesary padding

remove unnecesary styles

unnecesary styles

align menu items to left again

search results styles and fixes

prettier configuration fix

newline at end of file (variables.scss)

text color calculation for navbar

fix in img url

requested fixes

Transitions and focus fixes

fix in js settimeout

margin added in search-container

Fix in transition

Co-authored-by: Jairo Llopis <Yajo@users.noreply.github.com>

Remove duplicated css

ivantodorovich committed:
[IMP] Improve responsiveness in large screens: don't let the icons separate too much from each other

[FIX] Display of long search result items in mobile

[IMP] AppMenu searchbar style

[IMP] Add vertical padding to icons, to increase separation

[IMP] Add focus style on app icons

[IMP] Prevent user selection on menu

[IMP] Prevent dragging the menues

OCA-git-bot committed:
[UPD] README.rst

pedrobaeza committed:
[IMP] web_responsive: Adjust roadmap

* Removed one of the wishlist after #1654 and #1657
* Added the bug depicted in #1626

Closes #1626

OCA-git-bot committed:
[UPD] README.rst

Tardo committed:
[FIX] web_responsive: Hide search panel on small screens

Tardo committed:
[IMP] web_responsive: Cut long filters names in the filters menu

Tardo committed:
[IMP] web_responsive: Apply pre-commit changes

Tardo committed:
[FIX] web_responsive: Menu section overflow

Tardo committed:
[IMP] web_responsive: Bigger checkboxes in list view

ypapouin committed:
[FIX] web_responsive: Do not set overflow property on inline fields
Co-authored-by: Alexandre Díaz <alexandre.diaz@tecnativa.com>
Ensure that the title is displayed while the user holds the mouse click (while selecting content)

Commit:
[MIG] web_responsive: Migration to 14.0

Tardo committed:
[IMP] web_responsive: Scrollable dropdowns

oca-travis committed:
[UPD] Update web_responsive.pot

OCA-git-bot committed:
[UPD] README.rst

SplashS committed:
[FIX] error on grouped mobile kanban with zero groups

SplashS committed:
Added translation using Weblate (Russian)

SplashS committed:
Translated using Weblate (Russian)

Currently translated at 87.5% (21 of 24 strings)

Translation: web-14.0/web-14.0-web_responsive
Translate-URL: https://translation.odoo-community.org/projects/web-14-0/web-14-0-web_responsive/ru/

Yvesldff committed:
Added translation using Weblate (French (France))

Yvesldff committed:
Translated using Weblate (French (France))

Currently translated at 50.0% (12 of 24 strings)

Translation: web-14.0/web-14.0-web_responsive
Translate-URL: https://translation.odoo-community.org/projects/web-14-0/web-14-0-web_responsive/fr_FR/

SplashS committed:
[FIX] word wrap for appdrawer menu items

bosd committed:
Translated using Weblate (Dutch)

Currently translated at 77.7% (14 of 18 strings)

Translation: web-14.0/web-14.0-web_responsive
Translate-URL: https://translation.odoo-community.org/projects/web-14-0/web-14-0-web_responsive/nl/

remi-filament committed:
Translated using Weblate (French)

Currently translated at 77.7% (14 of 18 strings)

Translation: web-14.0/web-14.0-web_responsive
Translate-URL: https://translation.odoo-community.org/projects/web-14-0/web-14-0-web_responsive/fr/

saeed-raeisi committed:
Added translation using Weblate (Persian)

saeed-raeisi committed:
Translated using Weblate (Persian)

Currently translated at 100.0% (24 of 24 strings)

Translation: web-14.0/web-14.0-web_responsive
Translate-URL: https://translation.odoo-community.org/projects/web-14-0/web-14-0-web_responsive/fa/

saeed-raeisi committed:
Added translation using Weblate (Persian)

remi-filament committed:
Translated using Weblate (French)

Currently translated at 88.8% (16 of 18 strings)

Translation: web-14.0/web-14.0-web_responsive
Translate-URL: https://translation.odoo-community.org/projects/web-14-0/web-14-0-web_responsive/fr/

SplashS committed:
[MIG] web_responsive: Migration to 15.0

SplashS committed:
[IMP] show AppsMenu by default after login
also redirect to backend from website AppsMenu

SplashS committed:
[IMP] recaptured actual description screenshots

Tardo committed:
[IMP] web_responsive: Adapt label width

SplashS committed:
[IMP] big inputs on forms in edit mode

Tardo committed:
[FIX] web_responsive: Label width on modal

SplashS committed:
[FIX] make base odoo JS tests working with web_responsive module

Commit:
[UPD] Update web_responsive.pot

OCA-git-bot committed:
[UPD] README.rst

pedrobaeza committed:
[IMP] web_responsive: Indicate incompatibility

Closes #2149

tate11 committed:
Added translation using Weblate (Turkish)

SplashS committed:
[FIX] kanban mode for popup in Many2One fields on mobile

SplashS committed:
[FIX] redundant padding on kanban card

OCA-git-bot committed:
web_responsive 15.0.1.1.0

hieulucky111 committed:
[FIX]-web_responsive module: Go to wrong action menu when user is set with Home Action menu

OCA-git-bot committed:
web_responsive 15.0.1.1.1

jacekmichalski committed:
Added translation using Weblate (Polish)

jacekmichalski committed:
Translated using Weblate (Polish)

Currently translated at 91.6% (22 of 24 strings)

Translation: web-15.0/web-15.0-web_responsive
Translate-URL: https://translation.odoo-community.org/projects/web-15-0/web-15-0-web_responsive/pl/

SplashS committed:
[FIX] web_responsive: make keyboard navigation in Apps menu Search results works again.
Due to changes in javascript framework keyboard navigation was broken.
This commit makes it work again.

SplashS committed:
[FIX] web_responsive: attribute 'records_draggable' was no effect on desktop.
Kanban View has an attribute 'records_draggable', means 'whether it should be possible to drag records when kanban is grouped'.
It's support was broken.
Co-authored-by: MackZhong <mackz@sina.com>

OCA-git-bot committed:
web_responsive 15.0.1.1.2

Commit:
[UPD] Update web_responsive.pot

OCA-git-bot committed:
web_responsive 15.0.1.1.3

oca-transbot committed:
Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: web-15.0/web-15.0-web_responsive
Translate-URL: https://translation.odoo-community.org/projects/web-15-0/web-15-0-web_responsive/

vincent-hatakeyama committed:
Translated using Weblate (French)

Currently translated at 100.0% (24 of 24 strings)

Translation: web-15.0/web-15.0-web_responsive
Translate-URL: https://translation.odoo-community.org/projects/web-15-0/web-15-0-web_responsive/fr/

SplashS committed:
[FIX] web_responsive: fix dropdown caret visibility on website

before this commit dropdown on website doesn't have caret visible. F.x. to show option "Edit on backend"

after this commit caret will be visible

OCA-git-bot committed:
web_responsive 15.0.1.1.4

NICO-SOLUTIONS committed:
Translated using Weblate (German)

Currently translated at 54.1% (13 of 24 strings)

Translation: web-15.0/web-15.0-web_responsive
Translate-URL: https://translation.odoo-community.org/projects/web-15-0/web-15-0-web_responsive/de/

vincent-hatakeyama committed:
[FIX] web_responsive: only change width of main control panel

The Search More panel is incorrectly displayed due to the reduced screen
so only change width on the main screen.

Fixes #2395

OCA-git-bot committed:
web_responsive 15.0.1.1.5

bvanjeelharia committed:
[15.0][FIX] Fix problem of blank page when using web_responsive

Changes for default page open

OCA-git-bot committed:
web_responsive 15.0.1.1.6

bvanjeelharia committed:
[MIG] web_responsive: Migration to 16.0

Fix for optional dropdown checkbox in list view

Made Changes for compatibility with web_chatter_position module

Fix for attachment delete dialog

Commit:
[UPD] Update web_responsive.pot

OCA-git-bot committed:
[UPD] README.rst

weblate committed:
Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: web-16.0/web-16.0-web_responsive
Translate-URL: https://translation.odoo-community.org/projects/web-16-0/web-16-0-web_responsive/

weblate committed:
Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: web-16.0/web-16.0-web_responsive
Translate-URL: https://translation.odoo-community.org/projects/web-16-0/web-16-0-web_responsive/

tate11 committed:
Translated using Weblate (Turkish)

Currently translated at 87.5% (21 of 24 strings)

Translation: web-16.0/web-16.0-web_responsive
Translate-URL: https://translation.odoo-community.org/projects/web-16-0/web-16-0-web_responsive/tr/

bvanjeelharia committed:
[FIX] web_responsive: Align checkboxes in Favourites save current search view

bvanjeelharia committed:
[FIX] web_responsive: Align menu for dates in filters

OCA-git-bot committed:
web_responsive 16.0.1.0.1

bvanjeelharia committed:
[FIX] web_responsive: Show View Type Always

OCA-git-bot committed:
web_responsive 16.0.1.0.2

OCA-git-bot committed:
web_responsive 16.0.1.0.3

remi-filament committed:
Translated using Weblate (French)

Currently translated at 100.0% (18 of 18 strings)

Translation: web-16.0/web-16.0-web_responsive
Translate-URL: https://translation.odoo-community.org/projects/web-16-0/web-16-0-web_responsive/fr/

Hung412 committed:
[FIX] web_responsive: fix bug test ESLint

OCA-git-bot committed:
web_responsive 16.0.1.0.4

bvanjeelharia committed:
[FIX] web_responsive: Translation Button Position

OCA-git-bot committed:
web_responsive 16.0.1.0.5

hugosantosred committed:
[FIX] Split button templates to show translated labels

Fixes #2439

Commit:
[UPD] Update web_responsive.pot

OCA-git-bot committed:
web_responsive 16.0.1.1.0

weblate committed:
Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: web-16.0/web-16.0-web_responsive
Translate-URL: https://translation.odoo-community.org/projects/web-16-0/web-16-0-web_responsive/

badbole committed:
Translated using Weblate (Croatian)

Currently translated at 100.0% (18 of 18 strings)

Translation: web-16.0/web-16.0-web_responsive
Translate-URL: https://translation.odoo-community.org/projects/web-16-0/web-16-0-web_responsive/hr/

CarlosRoca13 committed:
[FIX] web_responsive: Wrap text when display_name is too long

CarlosRoca13 committed:
[IMP] web_responsive: Remove z and x hotkeys

OCA-git-bot committed:
web_responsive 16.0.1.1.1

OCA-git-bot committed:
web_responsive 16.0.1.1.2

NICO-SOLUTIONS committed:
Translated using Weblate (German)

Currently translated at 94.4% (17 of 18 strings)

Translation: web-16.0/web-16.0-web_responsive
Translate-URL: https://translation.odoo-community.org/projects/web-16-0/web-16-0-web_responsive/de/

Ivorra78 committed:
Translated using Weblate (Spanish)

Currently translated at 100.0% (18 of 18 strings)

Translation: web-16.0/web-16.0-web_responsive
Translate-URL: https://translation.odoo-community.org/projects/web-16-0/web-16-0-web_responsive/es/

OCA-git-bot committed:
[UPD] README.rst

chienandalu committed:
[IMP] web_responsive: public message chatter color

We give a color clue to the users so they can better tell if they're
sending an internal note or a public message.

Dislaimer: we use the `has()` pseudo class mainly because is simpler to
implement and the support is quite extended right now. Even Firefox will
catch up soon (now it has support behing a config flag).

So users with compatible browser will enjoy it, while others will just stay
as they were.

TT44953

kikopeiro committed:
Translated using Weblate (Spanish)

Currently translated at 100.0% (18 of 18 strings)

Translation: web-16.0/web-16.0-web_responsive
Translate-URL: https://translation.odoo-community.org/projects/web-16-0/web-16-0-web_responsive/es/

OCA-git-bot committed:
oca-github-bot post-merge updates

Commit:
[16.0][IMP] web_responsive: To follow the app menu icon in Odoo 16

OCA-git-bot committed:
[BOT] post-merge updates

Commit:
[16.0][FIX] web_responsive: Fixed full-screen apps menu

OCA-git-bot committed:
[BOT] post-merge updates

adrianojprado committed:
Translated using Weblate (Portuguese (Brazil))

Currently translated at 88.8% (16 of 18 strings)

Translation: web-16.0/web-16.0-web_responsive
Translate-URL: https://translation.odoo-community.org/projects/web-16-0/web-16-0-web_responsive/pt_BR/

adrianojprado committed:
Translated using Weblate (Portuguese (Brazil))

Currently translated at 100.0% (18 of 18 strings)

Translation: web-16.0/web-16.0-web_responsive
Translate-URL: https://translation.odoo-community.org/projects/web-16-0/web-16-0-web_responsive/pt_BR/

hafizhlf committed:
[FIX] web_responsive: Update z-index in apps menu.scss to fix overlapping issue:

- Increase z-index from 1000 to 10000 in apps_menu.scss.

OCA-git-bot committed:
[BOT] post-merge updates

hafizhlf committed:
Revert "[FIX] web_responsive: Update z-index in apps menu.scss to fix overlapping issue:"

This reverts commit 5de47a859287bbd71cd0532a739a0f943d6384ad.

hafizhlf committed:
[16.0][web_responsive] refactor: Update web_responsive.scss

- set z-index for .o_searchview_autocomplete element.

OCA-git-bot committed:
[BOT] post-merge updates

TesaLazarenko committed:
[MIG] web_responsive: Migration to 17.0

yajo committed:
[BUILD] web_responsive: resign as maintainer

Commit:
[UPD] Update web_responsive.pot

OCA-git-bot committed:
[BOT] post-merge updates

weblate committed:
Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: web-17.0/web-17.0-web_responsive
Translate-URL: https://translation.odoo-community.org/projects/web-17-0/web-17-0-web_responsive/

mymage committed:
Added translation using Weblate (Italian)

Ivorra78 committed:
Translated using Weblate (Spanish)

Currently translated at 100.0% (29 of 29 strings)

Translation: web-17.0/web-17.0-web_responsive
Translate-URL: https://translation.odoo-community.org/projects/web-17-0/web-17-0-web_responsive/es/

mymage committed:
Translated using Weblate (Italian)

Currently translated at 100.0% (29 of 29 strings)

Translation: web-17.0/web-17.0-web_responsive
Translate-URL: https://translation.odoo-community.org/projects/web-17-0/web-17-0-web_responsive/it/

TesaLazarenko committed:
[17.0][FIX] web_responsive: disable default file preview

OCA-git-bot committed:
[BOT] post-merge updates

Patelangel1995 committed:
[IMP] web_responsive: Redirect to home after login
Redirect to the home page after login will occur only if the user has enabled the 'Redirect to Home' configuration in their user profile settings

OCA-git-bot committed:
[BOT] post-merge updates

Commit:
[UPD] Update web_responsive.pot

OCA-git-bot committed:
[BOT] post-merge updates

weblate committed:
Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: web-17.0/web-17.0-web_responsive
Translate-URL: https://translation.odoo-community.org/projects/web-17-0/web-17-0-web_responsive/

mymage committed:
Translated using Weblate (Italian)

Currently translated at 100.0% (31 of 31 strings)

Translation: web-17.0/web-17.0-web_responsive
Translate-URL: https://translation.odoo-community.org/projects/web-17-0/web-17-0-web_responsive/it/

CarlosRoca13 committed:
[FIX] web_responsive: Can't access web using superuser

Before this change when trying to access web using superuser the web
interface is not charged.

By doing this change is all working normally.

Steps to reproduce the problem:

1. Log in
2. Go to web/become

OCA-git-bot committed:
[BOT] post-merge updates

victoralmau committed:
[FIX] web_responsive: Fix test related to ir.actions.act_window

Related to https://github.com/odoo/odoo/commit/cac20c5f2f0da7be6a75bd1281d528df7f34bfe6

Now it is necessary to have an ir.actions.act_window record

OCA-git-bot committed:
[BOT] post-merge updates

kafai-lam committed:
[FIX] web_responsive: Add support to serve menu icon in image url

Add support to serve menu icon in image URL format rather than base64.

It can take advantanges from browser caching and Nginx proxy.

Signed-off-by: Fai <lamkafai1997@gmail.com>

OCA-git-bot committed:
[BOT] post-merge updates

kobros-tech commented:
@lasley 
@danieltorres7 
@yajo 
@zeroheure 

I wonder why I see this error:

`File "/opt/odoo/odoo/sql_db.py", line 572, in execute
    return self._cursor.execute(*args, **kwargs)
  File "/opt/odoo/odoo/sql_db.py", line 354, in execute
    res = self._obj.execute(query, params)
psycopg2.errors.UndefinedColumn: column res_users.user_year_born does not exist
LINE 1: ...apps_menu_theme", "res_users"."is_redirect_home", "res_users...`

despite the tests succeed in a clean new database locally.
![Screenshot from 2024-12-24 21-16-11](https://github.com/user-attachments/assets/cefe8f0c-e5e1-4023-b2ae-7de7f2d65648)


I searched for this field user_year_born in odoo CE but it is not there, and I don't know where it came from.

kobros-tech commented:
The issue is inside web_pivot_computed_measure module tests.
I introduced a PR of a fix to the tests #3038

chaule97 commented:
Hi, I opened another PR https://github.com/OCA/web/pull/3057, welcome you review it

carlos-lopez-tecnativa committed:
[FIX] web_responsive: Make showAppsMenuItem work again

In Odoo 17, the Class TourStepUtils was removed, and stepUtils is now an object instead of a class. When this module is installed, the tours stop working and throw the error:
Uncaught Error: Dependencies should be defined by an array: function (require) {.

This commit adapts the code to align with these changes, ensuring that tours work correctly.

CarlosRoca13 commented:
Please, note this pr https://github.com/OCA/web/pull/3069

carlos-lopez-tecnativa commented on web_responsive/static/tests/test_patch.js:
```diff
@@ -0,0 +1,17 @@
+import {registry} from "@web/core/registry";
+import {stepUtils} from "@web_tour/tour_service/tour_utils";
+
+registry.category("web_tour.tours").add("test_patch", {
+    url: "/odoo",
+    steps: () => [
```
Take into account, this is not a test; it is a patch.

carlos-lopez-tecnativa commented on web_responsive/static/tests/test_patch.js:
```diff
@@ -0,0 +1,17 @@
+    steps: () => [
+        stepUtils.showAppsMenuItem(),
+        {
+            trigger: '.o_navbar_apps_menu"]',
```
This expression is wrong; it is missing brackets.

kobros-tech replied:
dear @CarlosRoca13 if this file is not correct, the test will fail and we will see red alarm by oca boat.

I don't believe if the syntax is wrong or the registry is not for tests, are you a developer?

we use pre-commit in OCA and this command is reviewing the syntax, as long as it is not raising errors here, then the syntax is correct too!

kobros-tech replied:
look at this properly and you will see that the file has run and succeeded:

https://github.com/OCA/web/actions/runs/12946365358/job/36110738186?pr=3037#step:8:297

pedrobaeza replied:
Everyone can make a mistake. I think you should reduce your tone with things like `are you a developer?`.

OCA-git-bot committed:
[BOT] post-merge updates

carlos-lopez-tecnativa commented on web_responsive/static/tests/test_patch.js:
```diff
@@ -0,0 +1,17 @@
+    steps: () => [
+        stepUtils.showAppsMenuItem(),
+        {
+            trigger: '.o_navbar_apps_menu"]',
```
> dear @CarlosRoca13 if this file is not correct, the test will fail and we will see red alarm by oca boat.
> 
> I don't believe if the syntax is wrong or the registry is not for tests, are you a developer?
> 
> we use pre-commit in OCA and this command is reviewing the syntax, as long as it is not raising errors here, then the syntax is correct too!

@kobros-tech  The original purpose of this code is to `patch` the original function of Odoo, not to add a new test. 
Therefore, this code is not executed when `pre-commit` is run or when tests are executed. It is used in other modules when you import `showAppsMenuItem`.

If you review the expression `.o_navbar_apps_menu"]`, it is syntactically incorrect. It should be:

```suggestion
            trigger: '.o_navbar_apps_menu',
```
Note: The extension of this filename must be renamed to` .esm.js.`

Sorry if I wrote parentheses instead of brackets; I edited my comment.

kobros-tech replied:
@carlos-lopez-tecnativa 

thanks for clarification :)

now for this file, I copied it from 17.0 to 18.0, maybe there was a change of it during migration in the original PR

please review and if there is any bug tell me.

carlos-lopez-tecnativa commented on eslint.config.cjs:
```diff
@@ -19,7 +19,7 @@ const config = [{
 
         ecmaVersion: 2024,
-        sourceType: "script",
+        sourceType: "module",
```
Why did you change this?

pedrobaeza replied:
Probably because he didn't rename to `.esm.js` the other file...

kobros-tech replied:
correct 👌

kobros-tech committed:
[IMP] web_responsive: pre-commit auto fixes

kobros-tech committed:
[18.0][MIG] web_responsive: Migration to 18.0

kobros-tech commented:
Wow wow wow
finally all tests succeed :)
@pedrobaeza

pedrobaeza commented:
I see that #3057 doesn't require to do anything in the tests on `web_pivot_computed_measure` to have a green CI. Can you check and try to not modify anything else?

pedrobaeza commented:
Testing the module in runboat, I detect that all the part that neutralizes the automatic preview seems to be not working (and consequently, the own sided preview).

kobros-tech commented:
> @lasley @danieltorres7 @yajo @zeroheure
> 
> I wonder why I see this error:
> 
> `File "/opt/odoo/odoo/sql_db.py", line 572, in execute return self._cursor.execute(*args, **kwargs) File "/opt/odoo/odoo/sql_db.py", line 354, in execute res = self._obj.execute(query, params) psycopg2.errors.UndefinedColumn: column res_users.user_year_born does not exist LINE 1: ...apps_menu_theme", "res_users"."is_redirect_home", "res_users...`
> 
> despite the tests succeed in a clean new database locally. ![Screenshot from 2024-12-24 21-16-11](https://private-user-images.githubusercontent.com/93363076/398464026-cefe8f0c-e5e1-4023-b2ae-7de7f2d65648.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MzgwNjI1MjgsIm5iZiI6MTczODA2MjIyOCwicGF0aCI6Ii85MzM2MzA3Ni8zOTg0NjQwMjYtY2VmZThmMGMtZTVlMS00MDIzLWIyYWUtN2RlN2YyZDY1NjQ4LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNTAxMjglMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjUwMTI4VDExMDM0OFomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTY5M2RlYzU3YTcyM2I4NGYyOTM3MjI3YzJjYzdmMGM3YTU1OGUxZmNkZTNjNTMxMjdlMzkxMTEzNWM2ZGFlMjEmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.ajBIVyyzt5Fgupq8hG-G2sIaV5P6aDFEO2i39Q-kROg)
> 
> I searched for this field user_year_born in odoo CE but it is not there, and I don't know where it came from.

@pedrobaeza 

what is the value of this conversation if the old tests is not interfering the migration?

even the other migration that you mentioned does not include all tests, but removed some of them.

if you want to see the errors with the other PR ask the contributor to leave the tests as they are and you will see this same error.

Even you can try it locally and catch the error.

I seek to correct not to mock the correctness

pedrobaeza commented:
OK, just asking. I always try to touch the minimum the existing things. Please check my other comment that affects the functionality.

kobros-tech commented:
> Testing the module in runboat, I detect that all the part that neutralizes the automatic preview seems to be not working (and consequently, the own sided preview).

we observed some errors in the UI we will fix, I don't know if they are the same but I am happy if you share them.

pedrobaeza commented:
I have already commented the problem. Is it not clear?

kobros-tech commented:
> I have already commented the problem. Is it not clear?

I will check it and compare to the issues we found, give me some time.

thanks for reviewing and helping me to improve the migration.

kobros-tech commented:
> OK, just asking. I always try to touch the minimum the existing things. Please check my other comment that affects the functionality.

nevermind, this seems complicated and worrying and shall see the value when all gaps get filled.

Thanks for inquiry.

kobros-tech commented:
> I have already commented the problem. Is it not clear?

Can you provide me with screenshots or log messages, we are fixing an error with the urls of search results

pilarvargas-tecnativa committed:
[IMP] web_responsive: Remove unnecessary calendar optimisation

Removed the optimisation of hiding the week number at a certain size
for consistency with the responsive design of Odoo enterprise. It is a
non-functional optimisation as it hides useful information that hardly
takes up any space. It also causes incompatibilities in the design with
extensions that do not take into account these specific changes.

TT54811

pilarvargas-tecnativa commented:
Please include https://github.com/OCA/web/pull/3075

kobros-tech commented:
I will add merged commits to this PR, thanks 
@kobros-tech

Ngquang commented:
Ping @pedrobaeza @carlos-lopez-tecnativa @pilarvargas-tecnativa any news guys

pilarvargas-tecnativa commented on web_responsive/static/src/views/form/status_bar_buttons.xml:
```diff
@@ -9,14 +9,14 @@
-        </xpath>
-        <xpath expr="//t[@t-set-slot='toggler']" position="replace">
+        </xpath> -->
+        <!-- <xpath expr="//t[@t-set-slot='toggler']" position="replace">
```
This is no longer necessary?

kobros-tech replied:
yes, thanks

ryanc-me committed:
[FIX] web_responsive: Fix blank screen on search (support menus w/o xmlid)

If a menu item is created manually via the Odoo UI, then it (likely) won't have an xmlid. As a result, the `t-foreach` fails, resulting in the whole Odoo UI crashing (a blank white screen is shown). This code should use `id` as the `t-key` instead, as that is *always* available, even for menu items created manually.

Ngquang commented:
ping @kobros-tech , any news man ?

kobros-tech commented:
PR #3105 is here too

kobros-tech commented:
@pedrobaeza @carlos-lopez-tecnativa @pilarvargas-tecnativa @Ngquang 

Dear gyps, I ask people in charge if @pedrobaeza can also work on having #3038 merged as this migration is dependednt on the improved tests cases there.
No need to create a demo model with fields while we can test already existing ones.

Please work on merging that PR so I can focus on add any value from #3057 with @chaule97 
His commit is here now and I am validating and work on keeping only the necessary changes, if any.

And please forgive me rage sometimes as I am used to be tough especially when I see the danger.

kobros-tech commented:
@chaule97 

I invite you to pull this branch: 
https://github.com/kobros-tech/web/tree/18.0-mig-web_responsive

or that one:
https://github.com/kobros-tech/web/tree/18.0-mig-web_responsive

and then add ONLY the necessary changes, we do migration as priority then add fixes that are against migrations

but entire chages they may not like here, and I in person I couldn't review all of your commit nicely as the changes are everywhere, and I assume the entire module is not bad to need changes everywhere.

carlos-lopez-tecnativa commented on web_responsive/static/src/components/apps_menu/apps_menu.xml:
```diff
@@ -34,8 +45,8 @@
                 class="o_grid_apps_menu__button"
-                title="Home Menu"
-                data-hotkey="h"
+                title="Home Menu XXX"
```
???

kobros-tech replied:
ok thanks

carlos-lopez-tecnativa commented on web_responsive/readme/DESCRIPTION.md:
```diff
@@ -0,0 +1,69 @@
+
+  ![image](../static/img/appsearch.gif)
+
+- Sticky header & footer in list view
+
+  ![image](../static/img/listview.gif)
```
This is now native to the Odoo server starting from V17, but I noticed that on mobile, the list view header is hidden. Is it possible to check this, or is this out of scope for this module?

![image](https://github.com/user-attachments/assets/de7650b9-36dd-4904-b9c9-6b88ca89aa6a)

kobros-tech replied:
let me check if it is related to that module or not

kobros-tech replied:
tell me which mobile type you use and which os version?

kobros-tech replied:
![Screenshot_2025-04-11-07-53-37-12_40deb401b9ffe8e1df2f1cc5ba480b12](https://github.com/user-attachments/assets/a7132a58-90f8-4a73-bf42-5928552c1a07)

carlos-lopez-tecnativa replied:
Sorry for the delay. I uploaded a video to explain my comment. Please let me know if this case can be handled by this module or not.

https://github.com/user-attachments/assets/aecf4654-27c6-4821-8f3e-30160656845b

carlos-lopez-tecnativa commented on web_responsive/readme/DESCRIPTION.md:
```diff
@@ -0,0 +1,69 @@
+**Features for mobile**: \* View type picker dropdown displays
+comfortably
+
+- Control panel buttons use icons to save space.
```
I got this error when trying to expand the action buttons.

![image](https://github.com/user-attachments/assets/b6e501cb-01b4-470f-bf7e-aceb41698da0)


![image](https://github.com/user-attachments/assets/f1e3a2cd-4d88-43b9-bf1d-0ecf88c43c59)


Additionally, I don't see the buttons.

![image](https://github.com/user-attachments/assets/43337ada-2efc-42fb-b152-1c206eb3e141)

kobros-tech replied:
all right

kobros-tech replied:
action button in 18.0 is containing smart buttons even after uninstalling the module 

I fixed xpath error as well

![Screenshot_2025-04-11-08-22-26-78_40deb401b9ffe8e1df2f1cc5ba480b12](https://github.com/user-attachments/assets/509b7846-0ab2-46c1-b840-3989aced92e3)

carlos-lopez-tecnativa commented on web_responsive/static/src/components/menu_odoo_searchbar/searchbar.scss:
This file is empty. Can it be removed?

carlos-lopez-tecnativa commented on web_responsive/static/tests/test_patch.js:
```diff
@@ -0,0 +1,25 @@
+/* Copyright 2021 ITerra - Sergey Shebanin
+ * Copyright 2025 Carlos Lopez - Tecnativa
+ * License LGPL-3.0 or later (http://www.gnu.org/licenses/lgpl). */
+odoo.define(
+    "web_responsive.test_patch",
+    ["@web_tour/tour_service/tour_utils", "@web/core/utils/patch"],
+    function (require) {
+        "use strict";
+
+        const {stepUtils} = require("@web_tour/tour_service/tour_utils");
+        const {patch} = require("@web/core/utils/patch");
```
Please convert to ESM

carlos-lopez-tecnativa reviewed (changes requested):
Functional review (tested locally)

pedrobaeza commented:
This migration still loses the preview display that is not opening automatically the attachments (for example in invoices), and that has the previous possibilities. Please restore that behavior.

kobros-tech commented:
> This migration still loses the preview display that is not opening automatically the attachments (for example in invoices), and that has the previous possibilities. Please restore that behavior.

yes, its a bug here
I will work on it

kobros-tech commented:
> This migration still loses the preview display that is not opening automatically the attachments (for example in invoices), and that has the previous possibilities. Please restore that behavior.

There was an issue with scss, on my side I saw chatter is taking 90% of the page I am still seeing document but in the big view only.

please check again if you don't see the document try to minimize the fonts or maximize the resolution.

chaule97 commented:
Hi @kobros-tech , I have created refactor code PR https://github.com/kobros-tech/web/pull/1 in your repo. Please review, merge and squash it

etobella commented:
@kobros-tech I made a basic functional test and everything looks great. What is missing?

kobros-tech commented:
> @kobros-tech I made a basic functional test and everything looks great. What is missing?

@etobella 
Thanks very much, I think nothing is missing after I compared my PR to another one.
I did changes requested by reviewers and merged some commits here as requested and merged in 17.0 too.

pedrobaeza reviewed (changes requested):
Please avoid this ugly view if no icon defined:

![imagen](https://github.com/user-attachments/assets/37b70e81-2f30-44d3-9f7a-09fbc48bfe27)

And this is still not covering the requirement of not auto-showing attachment for example in invoices:

![imagen](https://github.com/user-attachments/assets/e8a33d4c-37d9-4b0b-9223-7b193cf6e569)

and you have to use the other preview "style" where you can maximize it. Why are you ignoring me on this?

kobros-tech commented:
@pedrobaeza 

after I ran tests twice I confirm the timeout issue is solved by increasing the step_delay, reference:
https://github.com/odoo/odoo/blob/18.0/addons/im_livechat/tests/test_chatbot_form_ui.py#L42

PR:
https://github.com/OCA/web/pull/3158

kobros-tech commented:
@pedrobaeza 
good idea to add a default icon similar to modules don't have icons.

but for the preview, if I didn't understand you or I did and having it fine.

as if I zoom out screen it pops up and if I zoom in the previow goes away, and I will record a video to show.

kobros-tech commented:
@pedrobaeza 

I can't ignore you, but sometimes I misunderstand or unable to answer if I have a good answer.
I am sorry really.

kobros-tech commented:
### Preview in invoices screen:

https://github.com/user-attachments/assets/593e6725-d183-4df0-92cc-7ad1f85991dd

pedrobaeza commented:
I already stated in https://github.com/OCA/web/pull/3057#pullrequestreview-2721982131 and here in https://github.com/OCA/web/pull/3037#issuecomment-2795934987.

My initial comment with detailed explanation: https://github.com/OCA/web/pull/2405#issuecomment-1435661180

Extracted from there:

4. **Document viewer**

   Odoo includes natively the possibility of viewing in an embedded popup PDF files, videos or images. On previous versions, as the community edition always put the chatter on the bottom, these documents were previewed full size, covering the contents and not letting you to work side by side.

   ![Selección_016](https://user-images.githubusercontent.com/7165771/219865126-43de4b4f-596b-44a0-8c5d-8e4081ca8e20.png)


   `web_responsive` then included code for working on a more similar way than enterprise: putting the document viewer on the side, for easing the input for example on vendor bills, copying things side by side:

   ![Selección_017](https://user-images.githubusercontent.com/7165771/219865375-dfe9ce03-228e-43b7-b5c1-9c5c7d4f3eba.png)

   But it was also decided to do it in a more convenient way:

   - Don't opening by default the so considered main attachment by Odoo. This is not always what you want to do, and it covers the conversation (or move it to the bottom, being not consistent where to find it).
   - Allow to display this way any of the attachments, not only what they consider the main one.

   Odoo has now included these enterprises features in the community edition, but they use a different "document viewer" for the side view, and it still has the first mentioned flaw. So I think we need to tackle this in the OCA module. Possible options:

   1. Disable the "side viewer", and use the main document viewer in the same way as in previous versions.
   2. Don't move the chatter to bottom, don't open the side viewer in any case, opening it when clicking the attachments, and put a close X icon on the side viewer to "return" to the chatter.

----

See also #2745, and how it was added in 17 in #2795

pedrobaeza commented:
@kobros-tech I insist: the problem is the automatic display of that preview in the invoices. It should be disabled by default. We want the preview, but not automatic. Do you understand me?

pedrobaeza commented:
And that automatic preview is not using the same stuff:

![imagen](https://github.com/user-attachments/assets/a1066d38-7343-4487-b62f-e4760f8481c9)

kobros-tech commented:
let me understand this and see what I can do, at least to try.

kobros-tech commented:
@pedrobaeza 

I did try odoo without web_responsive and with it.

I saw the preview is poping up automatically, and in case of clicking the attachment the big screen of it pops up.

If I catch your point, I am asked to disable the default one but keeping the function if I click on attachmen?

becaue of the pros of the one we see when cliking on attachment over the other auto one?

If I am correct ok, I will try.

pedrobaeza commented:
> I am asked to disable the default one but keeping the function if I click on attachmen?

Yes. This is what we have as part of `web_responsive` module since a lot of versions ago, and what you have to keep migrating the module. It's not something new that is asked.

> becaue of the pros of the one we see when cliking on attachment over the other auto one?

Yes, already explained in quotes, but summarizing:

- Chatter visible without scrolling.
- Maximize possibility.
- Be able to see this way more than one attachment, not the main one.

pedrobaeza commented on web_responsive/static/src/components/apps_menu/apps_menu.xml:
```diff
@@ -35,7 +46,7 @@
                 class="o_grid_apps_menu__button"
                 title="Home Menu"
-                data-hotkey="h"
+                data-hotkey="shift+h"
```
Why are you changing this hotkey and making it more complicated?

pedrobaeza commented on web_responsive/static/src/views/form/form_statusbar.scss:
```diff
@@ -6,7 +6,7 @@
 
         .o_form_sheet {
-            overflow: auto;
+            overflow: unset;
```
Why changing this to `unset`?

pedrobaeza commented on web_responsive/static/tests/test_patch.esm.js:
```diff
@@ -5,8 +5,6 @@ odoo.define(
     "web_responsive.test_patch",
     ["@web_tour/tour_service/tour_utils", "@web/core/utils/patch"],
     function (require) {
-        "use strict";
```
Why removing this? It should be set always.

pedrobaeza commented:
/ocabot migration web_responsive

pedrobaeza reviewed (commented):
Functionally tested on runboat and now the part about not auto-previewing the attachment is working correctly.

Three comments inline for being clarified before proceeding with the merge, waiting also for some confirmation of the rest of the contributors/reviewers. cc @carlos-lopez-tecnativa @chaule97

carlos-lopez-tecnativa commented on web_responsive/static/src/components/apps_menu_item/apps_menu_item.xml:
```diff
@@ -0,0 +1,33 @@
+            t-att-data-menu-xmlid="props.app.xmlid"
+            t-att-href="props.href"
+            t-on-click="onClick"
+            draggable="false"
```
In Odoo enterprise, now it is possible to change the order of the menu items. I'm not sure if support for this can be added here, or if this attribute prevents its use.

https://github.com/user-attachments/assets/262b8cdf-df9c-4164-88e4-2659c9ab8631

carlos-lopez-tecnativa replied:
Ok for me, this comment can be considered a new feature.

carlos-lopez-tecnativa commented on web_responsive/static/tests/test_patch.esm.js:
```diff
@@ -0,0 +1,23 @@
+/* Copyright 2021 ITerra - Sergey Shebanin
+ * Copyright 2025 Carlos Lopez - Tecnativa
+ * License LGPL-3.0 or later (http://www.gnu.org/licenses/lgpl). */
+odoo.define(
```
Does `odoo.define` still have support? In my previous comment, I said to convert it to ESM, not just the file extension, but the code as well.

carlos-lopez-tecnativa reviewed (commented):
Tested and working. Just a few comments related to the features that work fine on desktop, and I think they can be improved to work well on mobile. Please let me know if they can be improved or if they are out of the scope of this module.

kobros-tech commented:
Are we ready for migration or is there any remaining comment?

pedrobaeza commented:
It seems there are 2 pending comments: https://github.com/OCA/web/pull/3037#discussion_r2052302234 and https://github.com/OCA/web/pull/3037#discussion_r2052290984

kobros-tech commented:
the two comments are not applied in version 17.0

kobros-tech commented:
### Differences between version 18.0 and version 17.0, open for opinions and comments:

![Screenshot from 2025-04-23 14-21-59](https://github.com/user-attachments/assets/30981a9e-f97a-43db-80f3-5a9ad61457a1)
![Screenshot from 2025-04-23 14-22-10](https://github.com/user-attachments/assets/0c22f2ac-b1cd-4266-ba61-dff861819cf1)
![Screenshot from 2025-04-23 14-22-20](https://github.com/user-attachments/assets/a42a3cd3-1a4e-4917-95c4-084e2c253264)
![Screenshot from 2025-04-23 14-22-35](https://github.com/user-attachments/assets/3a227525-c6ee-433f-a6dd-9fe0a560ba4e)
![Screenshot from 2025-04-23 14-22-52](https://github.com/user-attachments/assets/c3458202-ad44-4571-b9d5-f6ab0386f84d)
![Screenshot from 2025-04-23 14-23-11](https://github.com/user-attachments/assets/29a024e5-e8f2-49d0-bef5-8ee7fefea4b4)

kobros-tech commented:
Any specific migration comments?

I welcome migration related comments, but any new feature I can not add here now.

carlos-lopez-tecnativa reviewed (approved):
I wish the issue mentioned in this comment https://github.com/OCA/web/pull/3037#discussion_r2052290984 could be handled by this module, because it works fine on desktop, but in mobile view the header is not sticky.
But anyway, let's go!

kobros-tech commented:
it should be in a separate PR as the issue you mentioned is coming from old version, I see it as a feature in fact.

pedrobaeza reviewed (approved):
OK, let's go with this. Thanks for all the work.

/ocabot merge nobump

houssine78 commented:
thank you @pedrobaeza for having push the review so far !

thank you @kobros-tech for this most wanted PR
